### PR TITLE
Router: Add support for ../

### PR DESCRIPTION
**Pull request by vsavkin**

### CODE REVIEW

File: modules/@angular/router/src/directives/router_link.ts [hunk 1]
```diff
@@ -16,17 +16,17 @@ import {
...
 import {RouteSegment, UrlSegment, Tree} from '../segments';
-import {isString, isPresent} from '../facade/lang';
+import {isString, isArray, isPresent} from '../facade/lang';
 import {ObservableWrapper} from '../facade/async';
 
 @Directive({selector: '[routerLink]'})
 export class RouterLink implements OnDestroy {
   @Input() target: string;
-  private _changes: any[] = [];
+  private _comands: any[] = [];
```
Comment (arciisine) [inline modules/@angular/router/src/directives/router_link.ts]: Is this meant to be _commands?  Not sure if it's a typo or intentional.

**Comment by mprobst** [comment]
@vsavkin could you fix the typo and then rebase on master?

**Comment by vsavkin** [comment]
Rebased

File: modules/@angular/router/src/router.ts [hunk 2]
```diff
@@ -46,18 +58,94 @@ export class Router {
...
   }
 
+  /**
+   * Navigate based on the provided array of commands and a starting point.
+   * If no segment is provided, the navigation is absolute.
+   *
+   * ### Usage
+   *
+   * ```
+   * router.navigate(['team', 33, 'team', '11], segment);
```
Comment (todoubaba) [inline modules/@angular/router/src/router.ts]: ['team', 33, 'user', '11]

File: modules/@angular/router/src/directives/router_link.ts [hunk 3]
```diff
@@ -16,42 +16,81 @@ import {
...
+ *
+ * RouterLink expects the value to be an array of path segments, followed by the params
+ * for that level of routing. For instance `['/team', {teamId: 1}, 'user', {userId: 2}]`
+ * means that we want to generate a link to `/team;teamId=1/user;userId=2`.
+ *
+ * The first segment name can be prepended with `/`, `./`, or `../`.
+ * If the segment begins with `/`, the router will look up the route from the root of the app.
+ * If the segment begins with `./`, or doesn't begin with a slash, the router will
+ * instead look in the current component's children for the route.
+ * And if the segment begins with `../`, the router will go up one segment in the url.
```
Comment (zoechi) [inline modules/@angular/router/src/directives/router_link.ts]: Can there be more than one `../../user`?

**Pull request merged**